Increment required only when encryption is enabled

sharidas opened the pull request:
Increment to files in filecache is required only when
encryption is enabled. Else use the version as it is.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please fill out below information carefully.

Please note that any kind of change first has to be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.
-->

## Description
<!--- Describe your changes in detail -->
When encryption is not enabled then no need to go through each and every step in encryption wrapper's fopen. Instead we can get another storage wrapper's fopen handler.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/owncloud/core/issues/28780

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
This would help resolve the exceptions thrown in the logs.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- [x] Create external storage with SFTP
- [x] Create a folder `test` under shared folder.
- [x] upload a file under `test`
- [x] Delete the test folder
- [x] Try to access the `test` folder from the trash
- [x] No exception logs found

Also I have noted that during verification on my local machine the storage passes to local storage from the encryption wrapper ( when no encryption is enabled ).

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

PVince81 commented:
One reason why we still set the encryption wrapper even when encryption is disabled is mostly because we want to be able to detect formerly encrypted files and avoid sending these back.

So if we want to add such bypass we might need to at least put it after the encryption check, which might be when reading the header.

sharidas commented on lib/private/Files/Storage/Wrapper/Encryption.php:
```diff
@@ -937,9 +943,13 @@ protected function getHeader($path) {
 				// if the header was empty we have to check first if it is a encrypted file at all
 				// We would do query to filecache only if we know that entry in filecache exists
 				$info = $this->getCache()->get($path);
+				$encryptionModule = '';
```
Made alterations at the getHeader method. The intention here was to pass encryption module wrapped in array if encryption is enabled. And this value is later used https://github.com/owncloud/core/pull/28820/files#diff-938b56891f3079e628cfda1c331911b7R381 to call different storage's fopen. I hope this is the same header which is mentioned at https://github.com/owncloud/core/pull/28820#issuecomment-325367258

sharidas committed:
Increment required only when encryption is enabled

Increment to files in filecache is required only when
encryption is enabled. Else use the version as it is.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

SergioBertolinSG commented:
I see https://github.com/owncloud/core/issues/28780 fixed here.

PVince81 reviewed (approved):
:+1: code looks fine.

As discussed it would be good to adjust the trashbin tests to verify that the restored file's contents matches before deletion, and have this test run for encryption too (some encryption tests are disabled with trashbin)

Please also retest versions: restoring or downloading older versions of a file must still work on home and external storage

SergioBertolinSG reviewed (approved):
Integration tests part is ok

sharidas commented:
Verified that transfer ownership works with this change. Verified that upon deletion of the file after transfer, made some update, the versions are restored.

PVince81 commented:
@sharidas please backport to stable10

sharidas commented:
Backport : https://github.com/owncloud/core/pull/28880
